src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageReadmeControl.xaml.cs

```diff
@@ -17,54 +19,123 @@ namespace NuGet.PackageManagement.UI
     /// <summary>
     /// Interaction logic for PackageReadmeControl.xaml
     /// </summary>
-    public partial class PackageReadmeControl : UserControl
+    public partial class PackageReadmeControl : UserControl, IDisposable, INotifyPropertyChanged
     {
 #pragma warning disable CS0618 // Type or member is obsolete
         private IMarkdownPreview _markdownPreview;
 #pragma warning restore CS0618 // Type or member is obsolete
+        private bool _isDisposed = false;
+        private CancellationTokenSource _cancellationTokenSource;
+
+        private bool _isBusy = true;
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        public bool IsBusy
+        {
+            get => _isBusy || ReadmeViewModel.IsBusy;
+        }
+
+        public bool IsReadmeReady
+        {
+            get => !_isBusy && ReadmeViewModel.IsReadmeReady;
+        }
 
         public PackageReadmeControl()
         {
             InitializeComponent();
-            _markdownPreview = MarkdownPreviewSingleton.GetInstance();
-            descriptionMarkdownPreview.Content = _markdownPreview.VisualElement;
+            _cancellationTokenSource = new CancellationTokenSource();
         }
 
         public ReadmePreviewViewModel ReadmeViewModel { get => (ReadmePreviewViewModel)DataContext; }
 
+        public bool Initialize(IEditorOptionsFactoryService options)
+        {
+            if (options is null)
+            {
+                return false;
+            }
+            if (_markdownPreview is null)
+            {
+                // This class is marked as obsolete because the api hasn't been finalized, however we want to use IMarkdownPreview to maintain a centralized way of rendering markdown into html.
+#pragma warning disable CS0618 // Type or member is obsolete
+                var previewBuilder = new PreviewBuilder();
+#pragma warning restore CS0618 // Type or member is obsolete
+                previewBuilder.EditorOptions = options.GlobalOptions;
+                _markdownPreview = previewBuilder.Build();
+                descriptionMarkdownPreview.Content = _markdownPreview?.VisualElement;
+            }
+            return true;
+        }
+
         private void ReadmeViewModel_PropertyChanged(object sender, PropertyChangedEventArgs e)
         {
             if (e.PropertyName == nameof(ReadmePreviewViewModel.ReadmeMarkdown))
             {
-                NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(UpdateMarkdownAsync).PostOnFailure(nameof(PackageReadmeControl), nameof(ReadmeViewModel_PropertyChanged));
+                NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
```

Comment:
Give this was already there, don't feel like you need to fix this for this PR but this is another fire and forget that adds JTF overhead for no reason, see: https://github.com/NuGet/NuGet.Client/pull/6190#discussion_r1889485483.

@nkolev92 @zivkan